### PR TITLE
Support "Delete file" action for featured file attachments

### PR DESCRIPTION
**Pull request by leenagupte**
Trello: https://trello.com/c/qpeNzbYa

# What's changed?

Adds a 'Delete attachment' link to the featured attachment preview block. Clicking delete takes the user to a confirmation page which then redirects them to the current `/attachments` page, where they can see that the attachment is gone.

# Expected changes
Screenshot 2020-03-11 at 17 15 16
Screenshot 2020-03-11 at 17 17 52
Screenshot 2020-03-11 at 17 18 02

### CODE REVIEW

File: app/views/featured_attachments/_featured_attachment.html.erb [hunk 1]
```diff
@@ -1,4 +1,19 @@
+<% actions = [] %>
+<% actions << capture do %>
+  <%= form_tag(
+    file_attachment_path(document, attachment.file_attachment_id),
+    method: :delete,
+    class: "app-inline-block",
+    data: {
+      "modal-action": "delete",
```
Comment (benthorner) [inline app/views/featured_attachments/_featured_attachment.html.erb]: Minor: this is only applicable when this is used in a modal context. We can lose it here. It looks like it goes in a subsequent commit anyway.

File: app/controllers/file_attachments_controller.rb [hunk 2]
```diff
@@ -75,7 +75,7 @@ def destroy
     result = FileAttachments::DestroyInteractor.call(params: params, user: current_user)
     attachment_revision = result.attachment_revision
 
-    redirect_to file_attachments_path(params[:document]),
+    redirect_to params[:wizard] == "featured" ? featured_attachments_path(params[:document]) : file_attachments_path(params[:document]),
                 notice: t("file_attachments.index.flashes.deleted",
```
Comment (benthorner) [inline app/controllers/file_attachments_controller.rb]: We should keep translations specific to the screen on which they're shown. So in this case, the translation should be `attachments.index.flashes.deleted`, or similar. Could we use a separate one here, so they can vary independently? In this case, we still haven't confirmed which flashes for attachments, so we can actually leave it out entirely until that design work is completed.

Comment (leenagupte) [reply app/controllers/file_attachments_controller.rb]: I'll split up the logic, so that the flash message is only shown for inline attachments.

**Comment by leenagupte** [comment]
@benthorner Thanks for the comments. I've made the suggested changes. Do you want to re-review?

File: spec/features/editing_file_attachments/delete_file_attachment_spec.rb [hunk 3]
```diff
@@ -15,23 +25,45 @@ def given_there_is_an_edition_with_attachments
                       file_attachment_revisions: [@attachment_revision])
   end
 
+  def given_there_is_an_edition_with_featured_attachments
+    @attachment_revision = create(:file_attachment_revision, :on_asset_manager)
```
Comment (benthorner) [inline spec/features/editing_file_attachments/delete_file_attachment_spec.rb]: This shouldn't be necessary for this feature. I notice it's in the original scenario too - it should be safe to remove.

File: spec/features/editing_file_attachments/delete_file_attachment_spec.rb [hunk 4]
```diff
@@ -15,23 +25,45 @@ def given_there_is_an_edition_with_attachments
...
+
+    @edition = create(:edition,
+                      document_type: build(:document_type, :with_body, attachments: "featured"),
```
Comment (benthorner) [inline spec/features/editing_file_attachments/delete_file_attachment_spec.rb]: We shouldn't need the `with_body` trait here, as it only applies for scenarios involving a body field.

File: spec/features/editing_file_attachments/delete_file_attachment_spec.rb (hunk 3 above)
Comment (leenagupte) [reply spec/features/editing_file_attachments/delete_file_attachment_spec.rb]: Which bit?

File: spec/features/editing_file_attachments/delete_file_attachment_spec.rb (hunk 4 above)
Comment (leenagupte) [reply spec/features/editing_file_attachments/delete_file_attachment_spec.rb]: I followed the set up from `given_there_is_an_edition_with_attachments`. Does that mean that `:with_body` isn't needed there either?

File: spec/features/editing_file_attachments/delete_file_attachment_spec.rb (hunk 3 above)
Comment (benthorner) [reply spec/features/editing_file_attachments/delete_file_attachment_spec.rb]: Ah, sorry: `on_asset_manager`. It doesn't change the outcome from the feature's perspective - it doesn't have any Asset Manager stubs.

File: spec/features/editing_file_attachments/delete_file_attachment_spec.rb (hunk 4 above)
Comment (benthorner) [reply spec/features/editing_file_attachments/delete_file_attachment_spec.rb]: It is in that case: it's needed for scenarios involving a body field.

Comment (leenagupte) [reply spec/features/editing_file_attachments/delete_file_attachment_spec.rb]: I'm not sure I understand. Where is the body field being referenced in the "inline" tests?

**Comment by leenagupte** [comment]
@benthorner The tests have been updated and fixed up.

**Pull request merged**